compiler/Main.hs

```diff
@@ -79,21 +81,37 @@ compile opt (file:files) = runIdentity . flip evalNameyT firstName $ do
               desugared <- desugarProgram resolved
               infered <- inferProgram env desugared
               case infered of
-                Right (prog, env') ->
+                That (prog, env') ->
```

Comment:
[OK](https://i.kym-cdn.com/photos/images/newsfeed/001/053/259/ebd.jpg)

It might be better to shift this into a `([TypeError], Maybe a)` or something instead - I'll have a look at that another time.